lib/modules/manager/crossplane/extract.ts

```diff
@@ -0,0 +1,50 @@
+import { loadAll } from 'js-yaml';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { XPKGSchema } from './schema';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig?: ExtractConfig,
+): PackageFileContent | null {
+  // avoid parsing the whole file if it doesn't contain any resource having any pkg.crossplane.io/v*
+  if (!/apiVersion:\s+["']?pkg\.crossplane\.io\/v.+["']?/.test(content)) {
+    logger.trace({ packageFile }, 'No Crossplane package found in file.');
+    return null;
+  }
+
+  let list = [];
+  try {
+    list = loadAll(content);
+  } catch (err) {
+    logger.debug(
+      { err, packageFile },
+      'Failed to parse Crossplane package file.',
+    );
+    return null;
+  }
+
+  const deps: PackageDependency[] = [];
+  for (const item of list) {
+    const parsed = XPKGSchema.safeParse(item);
+    if (!parsed.success) {
+      logger.trace(
+        { item, errors: parsed.error },
+        'Invalid Crossplane package',
```

Comment:
```suggestion
        'Invalid Crossplane package.',
```

Maybe this needs a full stop at the end? Or is this a fragment that's going to be inserted into a proper sentence later?